src/osd/SnapMapper.cc

```diff
@@ -147,7 +147,7 @@ int OSDriver::get_next_or_current(
     ch, hoid, FuturizedStore::Shard::omap_keys_t{key}
   ).safe_then([&key, next_or_current] (FuturizedStore::Shard::omap_values_t&& vals) {
     assert(vals.size() == 1);
-    *next_or_current = std::make_pair(key, std::move(vals[0]));
```

Comment:
`omap_values_t` is:

```cpp
    using omap_values_t = std::map<std::string, ceph::bufferlist, std::less<>>;
```

so we definitely want the `begin()->second`.
Good catch, thanks!